packages/static-website/src/static-website.ts

```diff
@@ -102,6 +106,20 @@ export interface StaticWebsiteProps {
    */
   readonly distributionProps?: DistributionProps;
 
+  /**
+   * Custom bucket deployment properties.
+   *
+   * NOTE: sources and destinationBucket are required, however are overriden by construct specific instances. Please pass in the following as a minimum:
+   * ```ts
+   * {
+   *   sources: [],
+   *   // @ts-ignore
```

Comment:
nitt - not sure this is a good example. perhaps update with a more "real-life" example - ie: there should be no need for the // @ts-ignore in the example

Reply:
This is actually the way i'd want people to use it as otherwise they would need to create a bucket which doesn't get used. This is the problem with JSII atm as there is no way to do Partial without having to re-implement the props ourselves.

Reply:
On second thought, I have just forked `BucketDeploymentProps` and removed the required fields to make this simpler to use. Downside is more boilerplate code :(